functions/New-DbaAgentJob.ps1

```diff
@@ -0,0 +1,476 @@
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+https://dbatools.io/New-DbaAgentJob
+
+.EXAMPLE   
+New-DbaAgentJob -SqlServer 'sql1' -JobName 'Job1' -Description 'Just another job'
+Creates a job with the name "Job1" and a small description
+
+.EXAMPLE   
+New-DbaAgentJob -SqlServer 'sql1' -JobName 'Job1' -Disabled
+Creates the job but sets it to disabled
+
+.EXAMPLE
+New-DbaAgentJob -SqlServer 'sql1' -JobName 'Job1' -EventLogLevel 'OnSuccess'
+Creates the job and sets the notification to write to the Windows Application event log on success
+
+.EXAMPLE
+New-DbaAgentJob -SqlServer 'SSTAD-PC' -JobName 'Job1' -EmailLevel 'OnFailure' -EmailOperatorName 'dba'
+Creates the job and sets the notification to send an e-mail to the e-mail operator
+
+.EXAMPLE   
+New-DbaAgentJob -SqlServer 'sql1' -JobName 'Job1' -Description 'Just another job' -Whatif
+Doesn't create the job but shows what would happen.
+
+.EXAMPLE   
+"sql1", "sql2", "sql3" | New-DbaAgentJob -JobName 'Job1'
+Creates a job with the name "Job1" on multiple servers
+#>
+
+    [CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess = $true)]
+
+    param(
+        [parameter(Mandatory = $true, ValueFromPipeline=$true)]
+		[object[]]$SqlServer,
+        [Parameter(Mandatory = $false)]
+        [System.Management.Automation.PSCredential]$SqlCredential,
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$JobName,
+        [Parameter(Mandatory = $false)]
+        [switch]$Disabled,
+        [Parameter(Mandatory = $false)]
+        [string]$Description,
+        [Parameter(Mandatory = $false)]
+        [int]$StartStepId,
+        [Parameter(Mandatory = $false)]
+        [string]$CategoryName,
+        [Parameter(Mandatory = $false)]
+        [int]$CategoryId,
+        [Parameter(Mandatory = $false)]
+        [string]$OwnerLoginName,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet(0, "Never", 1, "OnSuccess", 2, "OnFailure", 3, "Always")]
+        [object]$EventLogLevel = '',
+        [Parameter(Mandatory = $false)]
+        [ValidateSet(0, "Never", 1, "OnSuccess", 2, "OnFailure", 3, "Always")]
+        [object]$EmailLevel = '',
+        [Parameter(Mandatory = $false)]
+        [ValidateSet(0, "Never", 1, "OnSuccess", 2, "OnFailure", 3, "Always")]
+        [object]$NetsendLevel = '',
+        [Parameter(Mandatory = $false)]
+        [ValidateSet(0, "Never", 1, "OnSuccess", 2, "OnFailure", 3, "Always")]
+        [object]$PageLevel = '',
+        [Parameter(Mandatory = $false)]
+        [ValidateScript({
+            if(($EmailLevel -in 1, "OnSuccess", 2, "OnFailure", 3, "Always") -and ($_.Length -ge 1))
+            {
+                $true
+            }
+            else
+            {
+                Throw "Please set the e-mail operator when the e-mail level parameter is set."
```

Comment:
We've gone away from throws since they are scary. @FriedrichWeinmann , can you please confirm that Stop-Function would be appropriate here?